src/core/dev-server/server.ts

```diff
@@ -166,6 +170,26 @@ export function createDevServer(nitro: Nitro): NitroDevServer {
   // Debugging endpoint to view vfs
   app.use("/_vfs", createVFSHandler(nitro));
 
+  // Apply headers based on routeRules. In cases where the dev server is being
+  // used, these would otherwise fail to be applied to public assets.
+  const routeRulesMatcher = toRouteMatcher(
+    createRadixRouter({ routes: nitro.options.routeRules })
+  );
+  app.use(
+    "/",
```

Comment:
I think we could add middleware only to static asset paths (and ideally only when matching them)

Reply:
I revisited this, and I was indeed missing an easier approach. The serveStatic setHeaders callback indeed gets passed the file path, but the response object can get at the original url path, so no need to do the rube-goldberg nonsense I was worrying about.